src/js/components/tweetsContainer.js

```diff
@@ -0,0 +1,114 @@
+import React from 'react';
+import TweetBlock from './tweetBlock';
+import {promiseTimeline, promiseTimelineSelf} from '../services/timelineService';
+
+export const DEFAULT_ERROR_MESSAGE = 'Something went wrong, please contact systems administrator.';
+
+export default class TweetsContainer extends React.Component {
+    constructor(props){
+        super(props);
+        this.state = {
+            tweets: [], // tweets retrieved 
+            isLoaded: false, // flags if request was loaded or not
+            error: null, // error message
+        };
+         
+        if(this.props.type in this.TYPE_SERVICE){ // change type to specified type, if specified type exists
+            this.serviceMethod = this.TYPE_SERVICE[this.props.type];
+            this.blockProps = this.TYPE_BLOCK_PROPS[this.props.type];
+        } else {
+            this.serviceMethod = this.TYPE_SERVICE['default']; // set type to default
+            this.blockProps = this.TYPE_BLOCK_PROPS['default'];
+        }
+    }
+
+    get TYPE_SERVICE(){
+        return {
+            default: () => promiseTimeline(),
+            self: () => promiseTimelineSelf().then(
+                (response) => {
+                    if(response?.data?.length == 0){ // if no tweets, set error message to special error message
+                        this.errorMessage = 'No tweets are available, post a tweet!';
+                        throw new Error('No tweets');
+                    } else {
+                        return response;
+                    }
+                }
+            )
+        };
+    }
+
+    get TYPE_BLOCK_PROPS(){
```

Comment:
if you're trying to make it generic, don't have different blocks of information as it'll get messy quickly